### DISCUSSION TITLE
Ability to change window opacity at the compositor level

### DISCUSSION THREAD

**Discussion by Reiddragon** [labwc/labwc#2359] [feature-requests]
Other Wayland compositors such as Kwin and Sway already allow changing the opacity of windows at the compositor level. This would be useful for changing the opacity of windows which don't support a transparent background by themselves, primarily for aesthetic reasons but also for the ability to have windows above something else without completely covering what's bellow (eg. keeping a chat window above a game window)

**Comment by Consolatis** [comment]
I've not checked if Openbox provides a similar feature but it seems this could be implemented with some `TransparencyIncrease` and `TransparencyDecrease` actions. Those would then call `wlr_scene_buffer_set_opacity()` for the window content itself (not sure how exactly that works with subsurfaces). I am not sure about server side window decorations, that would be way more complex to implement.

**Reply by Narrat** [reply]
> I've not checked if Openbox provides a similar feature (...)

IIRC there was none in that regard. Transparency/Opacity required compositors like `compton` and the likes to be run alongside `openbox`.

**Reply by Reiddragon** [reply]
> I've not checked if Openbox provides a similar feature but it seems this could be implemented with some `TransparencyIncrease` and `TransparencyDecrease` actions. Those would then call `wlr_scene_buffer_set_opacity()` for the window content itself (not sure how exactly that works with subsurfaces). I am not sure about server side window decorations, that would be way more complex to implement.

Window transparency has to be handled in the compositor, so Openbox doesn't have it as it wouldn't be the place to implement it, rather a compositor like picom or compton would have to provide it.

In Wayland, however, labwc is the compositor, so it's the place something like this would have to be implemented.
